File: Application/Tickets/TicketParams.cs
using Application.Core;
using Domain.Models;

namespace Application.Tickets;

public class TicketParams : PaginationParams
{
    public string? ProjectId { get; set; }
    public TicketStatus? Status { get; set; }
    public TicketPriority? Priority { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public string? SearchTerm { get; set; }
}

File: client/src/app/models/ticketParams.ts
export interface TicketParams {
    status: string | null;
    priority: number | null;
    startDate: string | null;
    endDate: string | null;
    orderBy: string;
    searchTerm: string | null;
    pageNumber: number,
    pageSize: number,
    projectId?: string | null
}

export interface ProjectPhaseTicketParams {
    phaseId: string | null;
    pageNumber: number,
    pageSize: number,
}

export class TicketParamsInitial implements TicketParams {
    status = null;
    priority = null;
    startDate = null;
    endDate = null;
    orderBy = "date";
    searchTerm = null;
    pageNumber = 1;
    pageSize = 10;

}